pypsa/networkclustering.py

```diff
@@ -351,14 +351,23 @@ def get_buses_linemap_and_lines(
     lines, linemap_p, linemap_n, linemap, lines_t = aggregatelines(
         network, buses, interlines, line_length_factor, with_time
     )
+    # network can be reduced to a set of isolated nodes in course of clustering (e.g. Rwanda)
+    if lines.empty:
+        lines_res = (
+            lines.drop(columns=["bus0", "bus1"])
+            .rename(columns={"bus0_s": "bus0", "bus1_s": "bus1"}, copy=False)
```

Comment:
The combination of these two lines could be replaced with `lines.drop(columns=["bus0_s", "bus1_s"])`

Reply:
Agree, for an empty dataframe it looks much better. Thanks a lot for a catch :)